### ISSUE TITLE
[3.6.1 and others] Leaving the raid makes ammo in your mag FIR

### ISSUE THREAD

**Issue by refringe** [sp-tarkov/server#863]
## **General information**:

**Project issue is for:** </br>
3.6.1, but seen it on 3.5.7, 3.5.8, 3.6.0 and fairly sure it'll be on 3.7.0

**Expected result**:</br>
Ammo in mag stays not-FIR

**Actual result**:</br>
Ammo in mag turns into FIR

## **How to reproduce**:
1. Take a gun with spare ammo in raid
2. Empty your mag (by firing, probably unloading works too)
3. Fill your mag with non-FIR ammo you brought
4. Insert it into your gun

Notes: if mag was not fully emptied, ammo inside will stay non-FIR. Haven't checked if
1. Which gun you have equipped in hands matters
2. If unloading the mag instead of firing works too
3. If you don't insert the mag into the gun makes ammo FIR too

## **Log files**:
I really don't wanna attach these cuz they are horrendous and heavily modded, and I'd rather save anyone the pain of looking on these, but here you go.

---
### Attachments
- [SPT-3.6.1.zip](https://dev.sp-tarkov.com/attachments/58e49a4f-6613-4696-ab21-1c85fc1b9632) (83 KiB)

---

[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88) by _**MusicManiac**_

**Comment by refringe** [comment]
1. Mag doesn't have to be inserted in the gun
2. It doesn't matter which gun you have in hands when extracting
3. Works with mags in rig, as long as they were loaded after being emptied
4. Doesnt matter if you empty mag by shooting or unloading.

TLDR if you load empty mag with non-FIR ammo you brought with you it becomes FIR after extraction.

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19001) by _**MusicManiac**_

**Comment by refringe** [comment]
Tried replicating with a fresh 3.6.1 install, in 5 raids on Factory without bots: been unable to at all replicate this with an 545 Short, 6L31 mags and BP/PP 5.45mm ammo. All ammo in the mags seem to stay non-FiR as intended.

Next time though, try to replicate it on a fresh install of the latest version of AKI, without any mods or alterations being installed. Says it in the notes that you were meant to read and confirm with before posting as a new issue, for the most minimal of courtesy to the AKI team:
- You must be able to replicate the issue with a fresh profile, while running *no* mods.

The below could be responsible, from a quick skim of the smaller logs.
>[Info   :   BepInEx] Loading [SERVPH's Tarky Menu 1.0]
[Info   :ModulePatch] Enabled patch infiniteammo

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19005) by _**Sister-Fister**_

**Comment by refringe** [comment]
Just did a fresh install and confirmed that bug is there. Have you followed all the steps in "How to preproduce"?

---
### Attachments
- [SPT-3.6.1-frsh.zip](https://dev.sp-tarkov.com/attachments/7181dc0a-de25-4200-a2f7-69a8bacd3b5d) (30 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19006) by _**MusicManiac**_

**Comment by refringe** [comment]
Yeah, just actually confirmed it by accident with the most recent run I just did, so huh. Think I've actually come across it before in older versions, but didn't think much of it considering a lot of the time I end up with mixed ammo mags that contain both non-FiR and FiR ammo when extracting.
I wasn't able to replicate it by firing the mag until it was dry and loading that mag up, but I did replicate it for a mag I marked as AP and forgot to load in the AP stack I brought before the run. Weird that.

Also another edgecase with this bug seems to be that it's corrected to being non-FiR ammo if you load the mag into a gun at any point, but if you don't then you get the bug, or at least that's what happened with my testing.

Cheers for uploading fresh install logs, those are crucial to have after all and rule out many other problems that modding might introduce by accident.

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19007) by _**Sister-Fister**_

**Comment by refringe** [comment]
Should be resolved with https://dev.sp-tarkov.com/SPT-AKI/Server/commit/0392d9f76ea9f774a6d20d3cd48018ac31d26ca9

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19011) by _**chomp**_

**Comment by refringe** [comment]
Fixed with release: https://discord.com/channels/875684761291599922/875707166244360254/1141449450468937758

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19016) by _**CZPZ**_

**Comment by refringe** [comment]
![image](https://dev.sp-tarkov.com/attachments/aea2e0a4-d373-4c41-b1fc-679ed5e4c2f7)
![image](https://dev.sp-tarkov.com/attachments/b3426c8e-3357-4c9a-9154-fccbad24da7d)

---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/aea2e0a4-d373-4c41-b1fc-679ed5e4c2f7) (2297 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/b3426c8e-3357-4c9a-9154-fccbad24da7d) (3296 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/88#issuecomment-19017) by _**CZPZ**_
